### PR TITLE
apply rate-limit-service.yaml from samples, when you try rate-limit task

### PR DESCRIPTION
**Pull request by johnzheng1975**
ratelimit service/deployment and redis service/deployment is necessary for this example.
However, this is not provided.
I tried and test pass, for new changes.

[ ] Configuration Infrastructure
[x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by johnzheng1975** [comment]
Change old PR https://github.com/istio/istio.io/pull/9140  to this

**Comment by ericvn** [comment]
@istio/wg-policies-and-telemetry-maintainers Would you verify this makes sense.

I do notice this page isn't tested. @johnzheng1975, I'm not sure if you want to add some testing for this work. You can see https://github.com/istio/istio.io/tree/master/tests.

**Comment by ericvn** [comment]
You can look at the lint Details and select the raw build text to see the failure.

```
level=info msg="Execution took 3m16.731740531s"
    ./en/docs/tasks/policy-enforcement/rate-limit/index.md
       65 | 1. Create ratelimit service/deployment and redis  

>> 1 spelling error found in 645 files
```

File: content/en/docs/tasks/policy-enforcement/rate-limit/index.md [hunk 1]
```diff
@@ -62,6 +62,12 @@ backend, is used below.
               requests_per_unit: 100
     {{< /text >}}
 
+1. Create ratelimit service/deployment and redis service/deployment.
```
Comment (bianpengyuan) [inline content/en/docs/tasks/policy-enforcement/rate-limit/index.md:65]: We'd need to emphasize that this is for demo purpose only. I am thinking about something like `Create a global rate limit service which implements Envoy's [rate limit service protocol](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ratelimit/v3/rls.proto). As a reference, a demo configuration can be found [here](https://github.com/istio/istio/blob/master/samples/ratelimit/rate-limit-service.yaml), which is based on a [reference implementation](https://github.com/envoyproxy/ratelimit) provided by Envoy.`

File: content/en/docs/tasks/policy-enforcement/rate-limit/index.md [hunk 2]
```diff
@@ -62,6 +62,8 @@ backend, is used below.
               requests_per_unit: 100
     {{< /text >}}
 
+1. Create a global rate limit service which implements Envoy's [rate limit service protocol](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ratelimit/v3/rls.proto). As a reference, a demo configuration can be found [here](https://github.com/istio/istio/blob/master/samples/ratelimit/rate-limit-service.yaml), which is based on a [reference implementation](https://github.com/envoyproxy/ratelimit) provided by Envoy.
```
Comment (bianpengyuan) [inline content/en/docs/tasks/policy-enforcement/rate-limit/index.md:65]:
```suggestion
1. Create a global rate limit service which implements Envoy's [rate limit service protocol](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ratelimit/v3/rls.proto). As a reference, a demo configuration can be found [here]({{< github_blob >}}/samples/ratelimit/rate-limit-service.yaml), which is based on a [reference implementation](https://github.com/envoyproxy/ratelimit) provided by Envoy.
```

**Review by bianpengyuan — APPROVED** [review]

**Comment by johnzheng1975** [comment]
Interesting, seems the fail is not caused by my change.
Is it a defect of test script? thanks.
```
2021-03-18T06:10:03.287194Z	info	tf	=== DONE (failed):  Test: 'tests_setup_profile_none[TestDocs/tasks/traffic-management/egress/egress-control/test.sh/test.sh] (3m40.606629834s)' ===
2021-03-18T06:10:37.329336Z	info	tf	=== DONE (failed):  Test: 'tests_setup_profile_none[TestDocs/tasks/traffic-management/egress/egress-control/test.sh] (4m14.880316344s)' ===FAIL
2021-03-18T06:12:42.412204Z	info	tf	=== FAILED: Test Run: 'tests_setup_profile_none' (exitCode: 1) ===
2021-03-18T06:12:42.414445Z	info	tf	=== Suite "tests_setup_profile_none" run time: 22m48.246076242s ===
FAIL	istio.io/istio.io/tests/setup/profile_none	1368.380s
FAIL
tests/tests.mk:29: recipe for target 'doc.test.profile_none' failed
```

**Comment by ericvn** [comment]
The two test failures (when I looked now) are known test failures waiting on a fix from Envoy/Istio related to timeouts not working. Hoping for a fix soon.

**Review by ericvn — APPROVED** [review]

**Comment by johnzheng1975** [comment]
Thanks @bianpengyuan  @ericvn 

Just kind reminder, 
https://github.com/istio/istio/blob/release-1.9/samples/ratelimit/rate-limit-service.yaml is 404.
https://github.com/istio/istio/blob/master/samples/ratelimit/rate-limit-service.yaml is good.

So, cherrypick/release-1.9 for this PR **may not work**, until also put https://github.com/istio/istio/pull/31333 in istio 1.9.

**Pull request merged**